crates/vm/src/system/memory/offline_checker/mod.rs

```diff
@@ -7,14 +7,16 @@ pub use bus::*;
 pub use columns::*;
 
 #[repr(C)]
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub struct MemoryReadAuxRecord {
     pub prev_timestamp: u32,
 }
 
 #[repr(C)]
-#[derive(Debug)]
-pub struct MemoryWriteAuxRecord<const NUM_LIMBS: usize> {
+#[derive(Debug, Clone)]
+pub struct MemoryWriteRecord<T, const NUM_LIMBS: usize> {
     pub prev_timestamp: u32,
-    pub prev_data: [u8; NUM_LIMBS],
+    pub prev_data: [T; NUM_LIMBS],
 }
+
+pub type MemoryWriteAuxRecord<const NUM_LIMBS: usize> = MemoryWriteRecord<u8, NUM_LIMBS>;
```

Comment:
I am not a fan of this. Aux here doesn't have anything to do with `u8`. You should be careful about naming.